impress-org/givewp-next-gen
# Refactor: Limit sections to top level blocks

Resolves #21

## Description



This PR restricts Section blocks to top level blocks and prevents nested sections.

To do this, field blocks are injected with a list of registered section blocks to allow as parent blocks. This enforces the top level section blocks.

## Affects



Fields and Sections are now configured as an exported array and registered as blocks in the main `index.js` entry point.

## Visuals



| Sections at top level | No nested sections |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/10858303/177630822-10ee9427-451c-4946-bb73-35cc62af182b.png) | ![image](https://user-images.githubusercontent.com/10858303/177630959-1649e204-299e-45a2-b77c-5e473ac751af.png) |

## Testing Instructions



- Should not see fields as top level blocks
- Should not see sections as nested blocks

## Pre-review Checklist
